### ISSUE TITLE
minor improvements to arg form validation UI

### ISSUE THREAD

**Issue by emlys** [natcap/invest#912]
Just some thoughts on improving how this looks:

<img width="400" alt="image" src="https://user-images.githubusercontent.com/43770515/159579260-6cf96e8d-a4a0-46b5-9a34-af59b1bf13c7.png">

- The first message should say "write access", not "w access"
- The message for a directory should say "access to this directory", not "this file"
- The input type should not be repeated in the validation message. Instead of "directory: Input is required but has no value", just say "Input is required but has no value"
- The message should be left-justified in line with the left edge of the input box
- The input label should be vertically centered relative to the input box, not including the validation message. As it currently is, when a validation message appears, the input label moves down to re-center itself relative to the row. That's a little visually distracting.
- It would look more consistent if the dropdown box was outlined in green, indicating that whatever selected option is valid.
- The dropdown arrow V is a little too close to the right edge of the box (not consistent with the other margins)
- It could be nice to use [switches](https://react-bootstrap.github.io/forms/checks-radios/#switches) instead of yes/no radio buttons for boolean inputs.

**Comment by davemfish** [comment]
>The input type should not be repeated in the validation message. Instead of "directory: Input is required but has no value", just say "Input is required but has no value"

I think the type hint is potentially useful to keep in the message. It's redundant when the form field is already showing it as the placeholder text, but it's not redundant if the field contains other text. 

>It could be nice to use [switches](https://react-bootstrap.github.io/forms/checks-radios/#switches) instead of yes/no radio buttons for boolean inputs.

I believe we already changed to switches

All the other suggestions sound like good ones to me!

**Comment by emilyanndavis** [comment]
> > The input type should not be repeated in the validation message. Instead of "directory: Input is required but has no value", just say "Input is required but has no value"
> 
> I think the type hint is potentially useful to keep in the message. It's redundant when the form field is already showing it as the placeholder text, but it's not redundant if the field contains other text.

This is a good point, and a great illustration of the potential pitfalls of using placeholders in form fields. As we decided together in our conversation on this topic, I've removed the input type from the validation message and instead added it to the label, where it will appear in parentheses, e.g., `Workspace (directory)` or `Area Of Interest (vector, _optional_)`. This will ensure this helpful info persists in the UI regardless of interaction state.

I've left the placeholders as they are, but we can certainly discuss updating or removing them if we find they're unnecessarily redundant!
